aries_vcx_core/src/ledger/indy_ledger.rs

```diff
@@ -6,22 +6,37 @@ use crate::{indy, PoolHandle, WalletHandle};
 use super::base_ledger::{AnoncredsLedgerRead, AnoncredsLedgerWrite, IndyLedgerRead, IndyLedgerWrite};
 
 #[derive(Debug)]
-pub struct IndySdkLedger {
+pub struct IndySdkLedgerRead {
     indy_wallet_handle: WalletHandle,
```

Comment:
perhaps can be removed now?

Reply:
Maybe not worth it doing it here, but how about taking a generic implementing `BaseWallet` along with the `submitter_did`? Something like `Option(&str, &T)` where `T: BaseWallet`. Shouldn't that remove the need to store the wallet handle while also making it easier to work with that scenario since the arguments are either provided together or not at all?